src/utils/narrow.js

```diff
@@ -437,7 +438,18 @@ export const apiNarrowOfNarrow = (
       // TODO(server-2.1): just send IDs instead
       return [{ operator: 'pm-with', operand: emails.join(',') }];
     },
-    search: query => [{ operator: 'search', operand: query }],
+    // The search narrow can behave in two ways,
+    // one where there is no filters, so the user can search a message in all the streams
+    // the second where the user wants to search in a particular stream
```

Comment:
These three lines are providing potentially helpful context on what we intend for these narrows to mean -- what the meaning is of a search narrow with `streamId` present or of one with it absent.

The most helpful place for that sort of information to live is on the type definition, instead of down here in the implementation of one of the functions that consumes the narrow. That way, it's in a central place that is natural to refer to whenever someone's trying to understand these objects.

Specifically, let's put it right above the line where you added `streamId` in the type definition.

Also, it's helpful for documentation like this to be concise. Try and see if you can find a way to express the essential content of what this is saying but in fewer words.